### DISCUSSION TITLE
Significant Lazy Component Loading Delays Due to Accumulated Click Requests Not Canceling Previous Ones

### DISCUSSION THREAD

**Discussion by Dev-RaheelAhmad** [livewire/livewire#6290] [1. Help]
Lazy components within Livewire face an issue where clicking a link multiple times can lead to a delay in loading the component. This delay arises because each click triggers a new request to the server, causing the component to await the completion of all requests before rendering. To tackle this, why not consider restricting or canceling previous requests when a user clicks on the link more than once?

Furthermore, it's advisable not to wait for the completion of prior requests and instead execute all user interactions concurrently. The speed at which each request will be processed is uncertain, so waiting for preceding requests might prove unnecessary. In situations like this, insisting on waiting for previous requests could potentially cause the application to freeze.

An effective approach would be to lock the wire:navigate feature if a user has already clicked it once and a request is in progress. This strategy prevents users from sending multiple requests and thereby delaying the loading of the lazy component. 

For instance, consider a scenario where a lazy component takes 2 seconds to load. If the link is clicked 10 times in quick succession, the component's loading time might extend to 20 seconds instead of the expected 2 seconds. This delay is attributed to the simultaneous sending of multiple requests, causing the application to process these 10 requests sequentially, rather than adopting the more efficient approach of canceling previous requests in favor of the current one.

**Comment by Dev-RaheelAhmad** [comment]
@calebporzio @joshhanley, kindly need your help... 🙏

**Reply by Cbrad24** [reply]
This functionality already existed in V2 for `wire:click` etc. You could queue up button clicks and Livewire would tackle them sequentially.

Not sure this would work for the new `wire:navigate` but what I used to do with buttons was add something like:
```php
<button wire:click="runClick" wire:target="runClick" wire:loading.attr="disabled">
Click Me
</button>
```
This would disable the buttons while a request was being sent. If you need to expand the scope you can adjust `wire:target`.

Hopefully this works with `wire:navigate` for you.

**Reply by Dev-RaheelAhmad** [reply]
@Cbrad24 I want to cancel the previous pending request after every new request triggered, not to disable the click during request! 
Also what about my other question here: https://github.com/livewire/livewire/discussions/6253 
I mentioned both of these 2 issues here: https://github.com/livewire/livewire/discussions/6422
